### DISCUSSION TITLE
Hinzufügen von flexiblen Stromtarifen?

### DISCUSSION THREAD

**Discussion by spcqike** [hoylabs/OpenDTU-OnBattery#306] [Ideas]
Hallo zusammen,

Zuletzt waren die Strompreise auf der Börse ja, wenn auch nur einen Tag, so tief, dass es wirtschaftlich war, die Einspeisung zu stoppen und Strom zu "kaufen". 
Da kommt die Frage auf, könnte man nicht flexible Stromtarife direkt in der OpenDTU berücksichtigen?

aWATTar bspw hat eine API, die man abfragen kann, da gibt es ab 14. Uhr die stündlichen Strompreise für den Folgetag.
https://api.awattar.de/v1/marketdata?start=1688940000000

Ohne Startwert gibt es die Strompreise für die kommenden 12 Stunden.

Ich könnte mir bspw. ein Option entweder am PowerMeter oder am Dynamischen Power Limiter vorstellen, der die Einspeisung ab einem einzustellenden Wert drosselt oder gar stoppt.

In meinem Fall (noch ohne Batterie) würde ich die Einspeisung gern automatisch auf 0 setzen (bzw. auf den im DPL bereits eingetragenen Wert), wenn der Strompreis auf der Börse 0 beträgt. Einfach, weil zu der Zeit bereits ausreichend Strom am Markt ist und so niemand vorhandene "Überschüsse" meinerseits kompensieren müsste. (Auch wenn die im Einzelnen den nicht riesig sind, könnte es in Summe schon ein etwas ausmachen, oder?)

mit vorhandener Batterie hat man i.d.R. eh keine Einspeisung, oder? (Außer mit Solarpasstrough, wenn die Batterie voll ist. Da würde ich ggf. gleichen Verhalten wie oben ohne Batterie sehen?) Dennoch kann es sich auch dann lohnen, flexible Tarife zu berücksichtigen. 
Im Falle von aWATTar kommen auf dem Börsenpreis ja noch 3% Aufschlag und zusätzliche 15,83Cent/kWh Entgelte drauf. Wenn also der Strompreis auf der Börse, wie zuletzt, unter -16Cent/kWh fällt, verdient man mit dem Bezug von Strom aus dem Stromnetz Geld (und trägt aktiv zur Stbilisierung bei). In dem Fall sollte der WR komplett gestoppt werden und nur Netzbezug laufen.

Noch weiter geschaut, könnte man doch auch soweit planen, dass, wenn man die Batterie per AC Charger laden kann, man die Batterie gezielt bis zum Eintreten des negativ Preises entladen wird und dann rein mit AC Ladung wieder geladen wird, der Inverter dabei ebenfalls aus ist.

Was meint ihr? Ist sowas möglich? Macht es Sinn? 

Die automatische Einschaltung vom DPL bei erreichen von 0Cent/kWh (Börsenpreis) würde wöchentlich mehrmals passieren. Das Deaktivieren des Wechselrichters (und ggf. AC-Aufladen der Batterie?) wohl eher selten. Das könnte man jedoch für den Winter bspw. gut anpassen und nutzen, im Sinne von "Lade die Batterie mit einem Strompreis von <=10 Cent/kWh", sodass auch bei kleineren Anlagen im Winter die Batterie hergenommen werden kann, um zu "Stoßzeiten" (06-08 Uhr und 18-20 Uhr) keinen teuren Strom kaufen zu müssen.

Aktivie Netzstabilisierung soll sich für die Verbraucher ja auszahlen, über günstige Strompreise 😄 

Grüße 
PS: Ja, das kann man alles mit SmartHome / NodeRed / ... auch außerhalb der OpenDTU-onBattery lösen, aber das könnte man mit dem DPL auch (wird im Falle der OpenDTU Community ja auch so gemacht, da gibt es den DPL ja nicht in der Firmware). Ich bin in so einem Fall aber ein Freund von "Wenn es geht, sollte es auf so wenig Geräten wie möglich laufen, damit es weniger Schnittstellen/Fehlerquellen gibt" 😄

**Comment by madmartin** [comment]
Hallo @spcqike ,

grundsätzlich teile ich Deine Einstellung

> "Wenn es geht, sollte es auf so wenig Geräten wie möglich laufen,

aber wenn ich mir das da oben so durchlese, und bedenke daß es in naher Zukunft wahrscheinlich ein Dutzend Anbieter geben wird bei denen es dynamische Stromtarife gibt, dann halte ich es für nicht sinnvoll das in OpenDTU-OnBattery abzubilden.
Zu unflexibel.
Da bist Du mit externen Programmen, die das dann per MQTT umsetzen, mit Sicherheit flexibler und vor allem viel einfacher beim Debuggen.

**Reply by spcqike** [reply]
naja, der Börsenstrompreis (Day-Ahead-Market) ist ja einer. der ist ja fest und sollte für alle Anbieter gleich sein. 

Die Berechnung ((Börsenstrompreis + X_%) + Y_Cent_Netzentgelte) sollte überall ähnlich sein. X und Y können abweichen und sollten dann als Vorgabe / Config anpassbar sein. nach dem was ich gelesen habe unterscheiden sich die Anbieter hauptsächlich in der Grundgebühr und im "drumherum", also die Art und Weise der Datenübertragung / Dongles und so weiter. 

Prinzipiell sollte es doch relativ einfach sein, den Börsenpreis über die API abzufragen, gern auch stündlich zu jeder vollen Stunde, und darauf aufbauend zu regeln, oder?

**Comment by jab42** [comment]
Hat sich hier iregdnwas getan?

BTW: Ist schon jemandem aufgefallen, dass die Börsenpreise seit der Umstellung auf 15 Minuten, nicht mehr (wesentlich) negativ werden?
Wird die Windkraft abgeregelt, um negativen Preis zu vermeiden?

**Reply by Manos1966** [reply]
>  die Börsenpreise seit der Umstellung auf 15 Minuten, nicht mehr (wesentlich) negativ werden?

meinst du jetzt in November?
In den Wintermonaten wirst du selten negative Preise sehen. Erst in Sommer wieder.

**Reply by jab42** [reply]
> In den Wintermonaten wirst du selten negative Preise sehen.

Ja, aber weshalb?
<img width="1230" height="810" alt="grafik" src="https://github.com/user-attachments/assets/bb19093a-fbd6-450c-9611-d5f6222488ec" />

Mir kann doch keiner erzählen, dass da über 14 Stunden der Preis ohne Abregelung nicht tiefer schwankt.

Die regeln doch die Windkraft ab, damit wir nicht mehr sparen.

Die 19,8 sind übrigens meine Netznebenkosten...

**Reply by spcqike** [reply]
> Die regeln doch die Windkraft ab, damit wir nicht mehr sparen.

"Die da oben!!!"

Warum sollte ein halbwegs vernünftiger Energieerzeuger seine Anlage durchlaufen lassen und negative Preise riskieren, also auch noch Geld dafür zahlen, dass sein Überschuss abgenommen wird? Kostenloser Strom ist gut und das Ziel. 

Negative Preise kommen unter anderem daher, dass Erzeuger feste Einspeisevergütungen bekommen, den Strom also nicht zum Börsenstrompreis verkaufen, und daher kein Interesse an Reduzierung haben. Das führt unweigerlich zu einem "Überangebot" und Erzeuger, die nicht weiter reduzieren können, aber dennoch produzieren müssen (Mindestlast an konventionellen Kraftwerken, bevor sie gänzlich abschalten und dann Stunden bis Tage brauchen um wieder ans Netz zu kommen) werden gezwungen, Geld für die Abnahme zu zahlen. Vereinfacht gesagt. (Der Strommarkt ist viel komplexer als das, der Preis setzt sich aus vielen Faktoren zusammen)


Das ist ja das gute an den Erneuerbaren allgemein. sie sind super flexibel und schnell regelbar. anders, als konventionelle Kraftwerke (die wir, stand heute, aber ebenfalls täglich brauchen)

btw. muss die Stromproduktion und die Erzeugung immer im Gleichgewicht sein. Wenn der Strom auf der Börse also nichts kostet und Kraftwerke dort bereits reduzieren, kann man ja trotzdem mehr Strom verbrauchen. Dann Erhöhen die Anlagen ihre Leistung auch wieder. Das Abregeln passiert ja nur, weil der günstige Strom nicht ausreichend verbraucht wird.


> Die 19,8 sind übrigens meine Netznebenkosten...

ja die ~20Cent Nebenkosten haben wir auch. daher lohnt sich ein dynamischer Tarif bei uns aktuell nicht, wir haben noch 0.28€/kWh Festtarif.

**Comment by Manos1966** [comment]
Ich erwaehne es nur: 
HA (HomeAssistant) kann die EPEX Preise automatisch einlesen.

Zusaetzlich gibt es eine Integration, die die EPEX Strompreise auslesen kann und darauf basierend die guenstige Zeit**(en)** berechnen  kann um die gewuenschte Energie**menge** vom Netz zu laden.


<img width="953" height="967" alt="image" src="https://github.com/user-attachments/assets/d8d70f00-481b-48eb-92aa-a576bdb4387d" />

**Comment by Manos1966** [comment]
> Ja, aber weshalb?

Es ist einfach zu erklaeren: Solar!

- Im Winter hat man deutlich weniger "Helligkeit-Stunden" als im Sommer 
https://github.com/hoylabs/OpenDTU-OnBattery/wiki/Battery%E2%80%90Tipps#what-size-battery-do-you-need-to-cover-the-night-hours
- Die Sonne liegt tiefer als im Sommer
- Das Wetter ist schlechter (dunkler)

In der Regel, eine PV-Anlage produziert am Tag nur 1/3 der Strommenge, die sie im Sommer produzieren wird.

Also, waehrend der Sommerzeit, haben wir drei Mal mehr Strom aus der PV-Anlagen als sonst. Und diese Ueberkapazitaet druckt den Preis nach unten.

Dies passiert nicht in der Winterzeit.
